pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -146,118 +146,245 @@ func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStart
 
 }
 
-// shouldEnsureSuggested checks if the exempt priority level exists and returns
-// whether the suggested flow schemas and priority levels should be ensured.
-func shouldEnsureSuggested(flowcontrolClientSet flowcontrolclient.FlowcontrolV1beta1Interface) (bool, error) {
-	if _, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(context.TODO(), flowcontrol.PriorityLevelConfigurationNameExempt, metav1.GetOptions{}); err != nil {
-		if apierrors.IsNotFound(err) {
-			return true, nil
-		}
-		return false, err
+const thisFieldManager = "api-priority-and-fairness-config-producer-v1"
+
+// fsAutoUpdateStrategyFunc accepts the current and desired FlowSchema configuration and determines whether an update is necessary.
+// current is the existing configuration stored.
+// desired is the desired spec we want stored.
+// On return, shouldUpdate is set to true if auto update is required and newObject will represent the new configuration to be stored.
+type fsAutoUpdateStrategyFunc func(current, desired *flowcontrolv1beta1.FlowSchema) (newObject *flowcontrolv1beta1.FlowSchema, shouldUpdate bool)
+
+// plAutoUpdateStrategyFunc accepts the current and desired PriorityLevelConfiguration configuration and determines whether an update is necessary.
+// current is the existing configuration stored.
+// desired is the desired spec we want stored.
+// On return, shouldUpdate is set to true if auto update is required and newObject will represent the new configuration to be stored.
+type plAutoUpdateStrategyFunc func(current, desired *flowcontrolv1beta1.PriorityLevelConfiguration) (*flowcontrolv1beta1.PriorityLevelConfiguration, bool)
+
+// auto-update strategy for the suggested FlowSchema objects
+func fsSuggestedUpdateStrategy(current, desired *flowcontrolv1beta1.FlowSchema) (*flowcontrolv1beta1.FlowSchema, bool) {
+	if current == nil || desired == nil {
+		return nil, false
+	}
+
+	updateSpec := shouldAttemptAutoUpdate(current)
+	annotationMatch := isAutoUpdateAnnotationMatch(current, updateSpec)
```

Comment:
The names here do not make the semantics clear.  The term "match" misled me; only on re-reading did I notice that it was _not_ comparing two different API objects.  I eventually came to understand that `annotationMatch` is true if and only if the boolean parse succeeded.  It would be clearer if that bit were simply returned from the first func that tries the parse and kept in a variable with a clearer name.

Reply:
I made an attempt to make it more clear:
```
	updateSpec := shouldUpdateSpec(current)
	updateAnnotationKey := shouldUpdateAnnotation(current, updateSpec)
```

`shouldUpdateSpec` is not needed by the manadatory strategy, but `shouldUpdateAnnotation` is. I kept them separate for this reason.